web/src/components/organisms/Settings/Integration/hooks.ts

```diff
@@ -0,0 +1,152 @@
+    id: integration.id,
+    name: integration.name,
+    description: integration.description,
+    logoUrl: integration.logoUrl,
+    developerId: integration.developerId,
+    developer: integration.developer,
+    iType: integration.iType,
+    config: {
+      token: integration.config?.token,
+      webhooks: integration.config?.webhooks,
+    },
+  });
+
+  const integrations = useMemo(() => {
+    return (data?.me?.integrations ?? [])
+      .map<Integration | undefined>(integration =>
+        integration ? fromIntegration(integration) : undefined,
+      )
+      .filter((integration): integration is Integration => !!integration);
+  }, [data?.me?.integrations]);
+
+  const workspaceIntegrationMembers = useMemo(() => {
+    return (workspace?.members ?? [])
+      .map<IntegrationMember | undefined>(member =>
+        member && member.__typename === "WorkspaceIntegrationMember" && member.integration
+          ? {
+              active: member.active,
+              integration: fromIntegration(member.integration),
+              integrationRole: member.integrationRole,
+              invitedById: member.invitedById,
+            }
+          : undefined,
+      )
+      .filter(
+        (integrationMember): integrationMember is IntegrationMember =>
+          !!integrationMember &&
+          integrationMember.integration.name.toLowerCase().includes(searchTerm ?? ""),
+      );
+  }, [workspace, searchTerm]);
+
+  const handleIntegrationConnectModalClose = useCallback(() => {
+    SetSelectedConnectionModalIntegration(undefined);
+    setIntegrationConnectModalShown(false);
+  }, []);
+
+  const handleIntegrationConnectModalOpen = useCallback(() => {
+    SetSelectedConnectionModalIntegration(undefined);
+    setIntegrationConnectModalShown(true);
+  }, []);
+
+  const handleIntegrationSettingsModalClose = useCallback(() => {
+    setIntegrationSettingsModalShown(false);
+  }, []);
+
+  const handleIntegrationSettingsModalOpen = useCallback((integrationMember: IntegrationMember) => {
+    console.log(integrationMember);
+
+    SetSelectedIntegrationMember(integrationMember);
+    setIntegrationSettingsModalShown(true);
+  }, []);
+
+  const handleConnectionModalIntegrationSelect = useCallback((integration: Integration) => {
+    SetSelectedConnectionModalIntegration(integration);
+  }, []);
+
+  const [addIntegrationToWorkspaceMutation] = useAddIntegrationToWorkspaceMutation();
+
+  const handleIntegrationConnect = useCallback(async () => {
+    if (!selectedConnectionModalIntegration || !workspaceId) return;
+    const integration = await addIntegrationToWorkspaceMutation({
+      variables: {
+        integrationId: selectedConnectionModalIntegration.id,
+        workspaceId,
+        role: GQLRole.Reader,
+      },
+    });
+    if (integration.errors || !integration.data?.addIntegrationToWorkspace) {
+      setIntegrationConnectModalShown(false);
+    }
+    setIntegrationConnectModalShown(false);
```

Comment:
The error if statement is redundant

Reply:
Added comment to add notification later, and prevented modal from closing in case of error (like the other one below).